tendrl/ceph_integration/manager/crud.py

```diff
@@ -1,389 +1,131 @@
-
-            return result
-        elif object_type == POOL:
-            return osd_map['pools']
-        elif object_type == CRUSH_RULE:
-            return osd_map['crush']['rules']
+            return tendrl_ns.state_sync_thread.request_delete(
+                POOL,
+                object_id
+            )
         elif object_type == CRUSH_NODE:
-            return osd_map['crush']['buckets']
-        elif object_type == CRUSH_TYPE:
-            return osd_map['crush']['types']
+            return tendrl_ns.state_sync_thread.request_delete(
+                CRUSH_NODE,
+                object_id
+            )
+        elif object_type == EC_PROFILE:
+            return tendrl_ns.state_sync_thread.request_delete(
+                EC_PROFILE,
+                object_id
+            )
         else:
             raise NotImplementedError(object_type)
 
-    def _dump_request(self, request):
-        """UserRequest to JSON-serializable form
-
-        """
-        return {
-            'id': request.id,
-            'state': request.state,
-            'error': request.error,
-            'error_message': request.error_message,
-            'status': request.status,
-            'headline': request.headline,
-            'requested_at': request.requested_at.isoformat(),
-            'completed_at':
-            request.completed_at.isoformat()
-            if request.completed_at else None
-        }
-
-    def get_request(self, request_id):
-        """Get a JSON representation of a UserRequest
-
-        """
-        try:
-            return self._dump_request(
-                self._manager.requests.get_by_id(request_id))
-        except KeyError:
-            raise NotFound('request', request_id)
-
-    def cancel_request(self, request_id):
-        try:
-            self._manager.requests.cancel(request_id)
-            return self.get_request(request_id)
-        except KeyError:
-            raise NotFound('request', request_id)
-
-    def list_requests(self, filter_args):
-        state = filter_args.get('state', None)
-        fsid = filter_args.get('fsid', None)
-        requests = self._manager.requests.get_all()
-        return sorted([self._dump_request(r)
-                       for r in requests
-                       if (
-            state is None or r.state == state
-        ) and (fsid is None or r.fsid == fsid)],
-            lambda a, b: cmp(b['requested_at'], a['requested_at']))
-
-    def server_get(self, fqdn):
-        return self._manager.servers.dump(self._server_resolve(fqdn))
-
-    def server_list(self):
-        return [
-            self._manager.servers.dump(
-                s
-            ) for s in self._manager.servers.get_all()
-        ]
-
-    def server_get_cluster(self, fqdn, fsid):
-        return self._manager.servers.dump_cluster(
-            self._server_resolve(fqdn), self._fs_resolve(fsid)
+    def delete_rbd(self, pool_id, rbd_name):
```

Comment:
delete() function takes the `id` of the entity to be deleted. In case of `rbd` we need to pass `pool_id` and `rnd_name` both so introduced one for the same. This is a typical case. I tried delete() with passing `pool_id` in attributes, but no help so finally had to add one.